helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -218,7 +218,7 @@ private static class AssignableReplicaWithScore implements Comparable<Assignable
           .containsKey(replica.getResourceName());
       _isInBaselineAssignment =
           clusterModel.getContext().getBaselineAssignment().containsKey(replica.getResourceName());
-      _replicaHash = Objects.hash(replica.toString(), clusterModel.getAssignableLogicalIds());
+      _replicaHash = Objects.hash(replica.toString(), clusterModel.getAssignableNodesForInstanceTag(replica.getResourceInstanceGroupTag()));
```

Comment:
Shall we comment here why we are using the list of instances when computing hash?

Reply:
Added comments.